proof/invariant-abstract/AARCH64/ArchKernelInit_AI.thy

```diff
@@ -58,45 +59,44 @@ lemma pptr_base_num:
   "pptr_base = 0x8000000000"
   by (simp add: pptr_base_def pptrBase_def canonical_bit_def)
 
-(* IRQ nodes occupy 11 bits of address space in this RISCV example state:
-   6 for irq number, 5 for cte_level_bits. *)
+(* IRQ nodes occupy 14 bits of address space in this AARCH64 example state:
+   9 for irq number, 5 for cte_level_bits. *)
```

Comment:
They're not used in that many places, but is it worth making constants for these numbers, to make it easier if we ever need to change them again?

Reply:
AFAICS it is only this one proof in which they are unfolded. Actually, only the sum (14) is unfolded, the `irq` type and `cte_level_bits` already abstract the actual numbers. It's just the comment that is trying to explain what is going on that mentions all of them. Might make sense to avoid the numbers in the comment as well, though. @Xaphiosis what do you think?

Reply:
It would be limited benefit, but we could define `irq_node_bits` to be `14` (via Value_Type) and add that comment above it, then not repeat `14` in this proof. Since we already have cte_level_bits, do we have a ready definition for the `5`? Is it `LENGTH(undefined::irq)`?
If we have defs for both of the pieces and can derive a number, then we can make a definition which will then update itself should any of the pieces change... which is probably never.
My feeling is, reluctantly, let's put in a definition, but leave it in ArchKernelInit_A.

Reply:
5 is `cte_level_bits`. I don't think we need a value type, we only use the 14 in `mask` in this one lemma, so that means a constant definition right above this lemma would be enough (it shouldn't leak outside `ArchKernelInit_AI`, it's not used anywhere else. We'd need to fiddle with the use site to unfold the constant accordingly so the lemma fits. Should be doable if we want it.

Reply:
Sorry I meant 9 not 5 for LENGTH(undefined::irq)

Reply:
For completeness and not as an argument about it being worth a constant, it looks like there's one use of the irq number in this lemma as well.

Reply:
Done. Slightly spookily, none of the other lemmas needed adjustments. Now I need to investigate why that is.

Reply:
Right, we don't need any of this, because the `irq_node` is the last object allocated in the init state, we don't need its size to prove non-overlap with other objects. More things to remove..